x-pack/plugins/fleet/server/routes/uninstall_token/handlers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TypeOf } from '@kbn/config-schema';
+
+import type { PolicyUninstallTokenMap } from '../../../common/types/models/uninstall_token';
+
+import { appContextService } from '../../services';
+import type { UninstallTokenServiceInterface } from '../../services/security/uninstall_token_service';
+import type { FleetRequestHandler } from '../../types';
+import type { GetUninstallTokensResponse } from '../../../common/types/rest_spec/uninstall_token';
+import type { GetUninstallTokensRequestSchema } from '../../types/rest_spec/uninstall_token';
+
+export const getUninstallTokensHandler: FleetRequestHandler<
+  unknown,
+  TypeOf<typeof GetUninstallTokensRequestSchema.query>
+> = async (context, request, response) => {
+  const { agentTamperProtectionEnabled } = appContextService.getExperimentalFeatures();
+  if (!agentTamperProtectionEnabled) {
```

Comment:
What about not registering the route instead? Same think for the OpenApi spec, is there a way to not define it depending on the FF?

Reply:
sure! here's the solution for not registering the route: 69cbf5d

for OpenAPI, I couldn't find a viable solution to hide it conditionally, so maybe it's easier to remove it, and add it again when we release the feature. what do you think?

Reply:
@gergoabraham What do you think about doing it at the `routes/index.ts`? So you don't have to pass the config through the register function?
Something like:

```typescript
  const experimentalFeatures = parseExperimentalConfigValue(config.enableExperimental);
// [...]
  if (experimentalFeatures.agentTamperProtectionEnabled) { 
    registerUninstallTokenRoutes(fleetAuthzRouter);
  }
// [...]
```

As it is done for the other conditional config routes.

About OpenAPI, I would like to get fleet folks thoughts on it - @nchaulet

Reply:
@dasansol92  actually there is no unified approach, there are other route register functions that receive the `config`. I chose this solution because it was easier to test in the existing test file instead of adding a new test file for the `routes/index.ts`. let me know if this is okay with you